### PR TITLE
Only storing received changes in WriterProxy [6159]

### PR DESCRIPTION
**Pull request by MiguelCompany**
Reducing physical memory consumption by keeping only the sequence number of the received changes on WriterProxy

### CODE REVIEW

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/179/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/164/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/178/)

File: src/cpp/rtps/reader/WriterProxy.cpp [hunk 1]
```diff
@@ -330,83 +222,70 @@ bool WriterProxy::received_change_set(
...
-            {
-                ChangeFromWriter_t& ch = const_cast<ChangeFromWriter_t&>(*chit);
-                ch.setStatus(RECEIVED);
-                ch.setRelevance(is_relevance);
-            }
-            else
-            {
-                return false;
-            }
+            changes_received_.erase(seq_num);
```
[the remark this answers is not in the document]
Comment (MiguelCompany) [reply src/cpp/rtps/reader/WriterProxy.cpp]: done

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/195/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/178/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/194/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/205/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/187/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/204/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/224/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/203/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/223/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/232/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/211/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/231/)

**Comment by richiware** [comment]
Build status:
* Linux [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Linux/239/)
* Mac [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Mac/218/)
* Windows [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Manual%20Windows/238/)

**Comment by raquelalvarezbanos** [comment]
Failed tests are unrelated.

**Comment by cottsay** [comment]
This change caused a regression in our communication tests between Fast-RTPS and Connext. We're tracking the CI regression with ros2/build_cop#230

The new assert which is failing is https://github.com/eProsima/Fast-RTPS/pull/646/files#diff-f74e057b5a456d6378532071b5e0f27bR338

Please let me know if there is anything further I can do to help arrive at a resolution.

**Pull request merged**